src/include/ipc4/base-config.h

```diff
@@ -0,0 +1,130 @@
+ * original so it's easier to track changes with IPC host code.
+ */
+
+/**
+ * \file include/ipc4/base-config.h
+ * \brief IPC4 global definitions.
+ * NOTE: This ABI uses bit fields and is non portable.
+ */
+
+#ifndef __SOF_IPC4_BASE_CONFIG_H__
+#define __SOF_IPC4_BASE_CONFIG_H__
+
+#include <stdint.h>
+
+enum ipc4_sampling_frequency {
+	IPC4_FS_8000HZ = 8000,
+	IPC4_FS_11025HZ = 11025,
+	IPC4_FS_12000HZ = 12000, /**< Mp3, AAC, SRC only */
+	IPC4_FS_16000HZ = 16000,
+	IPC4_FS_18900HZ = 18900, /**< SRC only for 44100 */
+	IPC4_FS_22050HZ = 22050,
+	IPC4_FS_24000HZ = 24000, /**< Mp3, AAC, SRC only */
+	IPC4_FS_32000HZ = 32000,
+	IPC4_FS_37800HZ = 37800, /**< SRC only for 44100 */
+	IPC4_FS_44100HZ = 44100,
+	IPC4_FS_48000HZ = 48000, /**< Default */
+	IPC4_FS_64000HZ = 64000, /**< AAC, SRC only */
+	IPC4_FS_88200HZ = 88200, /**< AAC, SRC only */
+	IPC4_FS_96000HZ = 96000, /**< AAC, SRC only */
+	IPC4_FS_176400HZ = 176400, /**< SRC only */
+	IPC4_FS_192000HZ = 192000, /**< SRC only */
+	IPC4_FS_INVALID
+};
+
+enum ipc4_bit_depth {
+	IPC4_DEPTH_8BIT	= 8, /**< 8 bits depth */
+	IPC4_DEPTH_16BIT = 16, /**< 16 bits depth */
+	IPC4_DEPTH_24BIT = 24, /**< 24 bits depth - Default */
+	IPC4_DEPTH_32BIT = 32, /**< 32 bits depth */
+	IPC4_DEPTH_64BIT = 64, /**< 64 bits depth */
+	IPC4_DEPTH_INVALID
+};
+
+enum ipc4_channel_config {
+	IPC4_CHANNEL_CONFIG_MONO = 0,		/**< one channel only */
+	IPC4_CHANNEL_CONFIG_STEREO = 1,		/**< L & R */
+	IPC4_CHANNEL_CONFIG_2_POINT_1 = 2,	/**< L, R & LFE; PCM only */
+	IPC4_CHANNEL_CONFIG_3_POINT_0 = 3,	/**< L, C & R; MP3 & AAC only */
+	IPC4_CHANNEL_CONFIG_3_POINT_1 = 4,	/**< L, C, R & LFE; PCM only */
+	IPC4_CHANNEL_CONFIG_QUATRO = 5,		/**< L, R, Ls & Rs; PCM only */
+	IPC4_CHANNEL_CONFIG_4_POINT_0 = 6,	/**< L, C, R & Cs; MP3 & AAC only */
+	IPC4_CHANNEL_CONFIG_5_POINT_0 = 7,	/**< L, C, R, Ls & Rs */
+	IPC4_CHANNEL_CONFIG_5_POINT_1 = 8,	/**< L, C, R, Ls, Rs & LFE */
+	IPC4_CHANNEL_CONFIG_DUAL_MONO = 9,	/**< one channel replicated in two */
+	/**< Stereo (L,R) in 4 slots, 1st stream: [ L, R, -, - ] */
+	IPC4_CHANNEL_CONFIG_I2S_DUAL_STEREO_0 = 10,
+	/**< Stereo (L,R) in 4 slots, 2nd stream: [ -, -, L, R ] */
+	IPC4_CHANNEL_CONFIG_I2S_DUAL_STEREO_1 = 11,
+	IPC4_CHANNEL_CONFIG_7_POINT_1 = 12,	/**< L, C, R, Ls, Rs & LFE., LS, RS */
+	IPC4_CHANNEL_CONFIG_INVALID
+};
+
+enum ipc4_interleaved_style {
+	IPC4_CHANNELS_INTERLEAVED = 0,
+	IPC4_CHANNELS_NONINTERLEAVED = 1,
+};
+
+enum ipc4_sample_type {
+	IPC4_TYPE_MSB_INTEGER = 0,	/**< integer with Most Significant Byte first */
+	IPC4_TYPE_LSB_INTEGER = 1,	/**< integer with Least Significant Byte first */
+	IPC4_TYPE_SIGNED_INTEGER = 2,	/**< signed integer */
+	IPC4_TYPE_UNSIGNED_INTEGER = 3, /**< unsigned integer */
+	IPC4_TYPE_FLOAT = 4		/**< unsigned integer */
+};
+
+enum ipc4_stream_type {
+	IPC4_STREAM_PCM = 0,	/**< PCM stream */
+	IPC4_STREAM_MP3 = 1,	/**< MP3 encoded stream */
+	IPC4_STREAM_AAC = 2,	/**< AAC encoded stream */
+	IPC4_STREAM_MAX = 3,
```

Comment:
general usage is :  stream < IPC4_STREAM_MAX

Reply:
So this will need revisited (as we will likely have ti support more than just AAC and MP3) prior to the first release that uses this IPC major version. Can you mark this with a ```TODO: revisit max stream type``` comment.
This means the IPC will work as expected today with Windows and we can update Windows drivers at the same time.

Reply:
I also think MAX might be a bit confusing. COUNT is easier to understand. Reason being is we work with companies like Maxim and amps prefixed with Max. I would recommend confusing that term with codecs.

Reply:
sure, I will refine it